wandb/plot/confusion_matrix.py

```diff
@@ -37,7 +37,7 @@ def confusion_matrix(preds=None, y_true=None, class_names=None):
             class_names
         ), "Higher label class index than number of classes"
     else:
-        n_classes = max(max(preds), max(y_true))
```

Comment:
I think we need to evaluate this scenario carefully. Correctness here depends on whether or not the user is actually using "0" as a label. E.g., assume they just log one label "0": then max(max(0), max(0)) = 0 + 1 = 1 n_classes, correct. But if they just log one label "1", or basically any set of 1-indexed instead of 0-indexed classes, then max(max(1), max(1)) = 1 + 1 = 2 n_classes, incorrect. So I don't think we can do something that neat here. What do you think about taking the union( set(preds)), set(y_true)) and then the cardinality of that?

Reply:
Good point, this requires that its 0-indexed, but I can add that to the function doc strings

Reply:
This shouldn't be an issue anymore, since we have the mapping in place, the indexes can be anything sortable